api/envoy/extensions/rbac/principals/mtls_authenticated/v3/mtls_authenticated.proto

```diff
@@ -0,0 +1,31 @@
+syntax = "proto3";
+
+package envoy.extensions.rbac.principals.mtls_authenticated.v3;
+
+import "envoy/extensions/transport_sockets/tls/v3/common.proto";
+
+import "udpa/annotations/status.proto";
+import "validate/validate.proto";
+
+option java_package = "io.envoyproxy.envoy.extensions.rbac.principals.mtls_authenticated.v3";
+option java_outer_classname = "MtlsAuthenticatedProto";
+option java_multiple_files = true;
+option go_package = "github.com/envoyproxy/go-control-plane/envoy/extensions/rbac/principals/mtls_authenticated/v3;mtls_authenticatedv3";
+option (udpa.annotations.file_status).package_version_status = ACTIVE;
+
+// [#protodoc-title: RBAC MTls Authenticated Principal]
+// [#extension: envoy.rbac.principals.mtls_authenticated]
+
+// Authentication attributes for a downstream mTLS connection. All modes require that a peer certificate
+// was presented and validated using the ValidationContext in the DownstreamTlsContext configuration.
+message Config {
+  oneof type {
```

Comment:
As per https://github.com/envoyproxy/envoy/blob/main/api/STYLE.md, please prefer defining preference order over `oneof` for forward-compatibility reasons.

Reply:
What I was trying to accomplish was to make it so that it requires an affirmative configuration choice to not supply any `san_matcher`, as this is sometimes an accidental omission which could lead to a security issue, and using the oneof allowed it to be required by pgv validation rules, catching it early.

I had forgotten about the forward-compat issue. I think I can accomplish the same thing, but not depend on pgv-validation and instead have an error when loading config if nothing is configured.